File: src/main/java/com/sideproj/ratelimiter/configuration/Endpoint.java
package com.sideproj.ratelimiter.configuration;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;

@AllArgsConstructor
@NoArgsConstructor
@Data
public class Endpoint {
    public enum Strategy {
        TOKEN_BUCKET,
        LEAKING_BUCKET,
        FIXED_WINDOW_COUNTER,
        SLIDING_WINDOW_LOG,
        SLIDING_WINDOW_COUNTER
    }

    public enum LimitBy {
        ALL_REQUESTS,
        IP,
    }

    private String path;
    private int numberOfRequests;
    private int window;
    private Strategy strategy;
    private LimitBy limitBy;
    private ArrayList<Endpoint> endpoints;
}

File: configuration/endpoint.go
package configuration

type LimitingStrategy string
type LimitByCriteria string

const (
	TokenBucket          LimitingStrategy = "tokenBucket"
	LeakingBucket        LimitingStrategy = "leakingBucket"
	FixedWindowCounter   LimitingStrategy = "fixedWindowCounter"
	SlidingWindowLog     LimitingStrategy = "slidingWindowLog"
	SlidingWindowCounter LimitingStrategy = "slidingWindowCounter"
)

const (
	AllRequests LimitByCriteria = "allRequests"
	IP          LimitByCriteria = "ip"
)

type Endpoint struct {
	Path             string           `json:"path"`
	NumberOfRequests uint64           `json:"numberOfRequests"`
	Window           uint64           `json:"window"`
	Strategy         LimitingStrategy `json:"strategy"`
	LimitBy          LimitByCriteria  `json:"limitBy"`
	Endpoints        []Endpoint       `json:"endpoints"`
}
